File: public/examples/integrations/mcp-servers/stripe_api/list_payment_method_domains_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "StripeApi.ListPaymentMethodDomains";

// Start the authorization process
const authResponse = await client.tools.authorize({tool_name: TOOL_NAME});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "domain_name": "payments.example.com",
  "pagination_cursor_ending_before": "cursor_abc123",
  "fields_to_expand": [
    "webhooks",
    "settings"
  ],
  "limit_number_of_returned_objects": 25,
  "pagination_cursor_starting_after": "cursor_def456",
  "include_enabled_domains": true
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));

File: public/examples/integrations/mcp-servers/stripe_api/list_payment_method_domains_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "StripeApi.ListPaymentMethodDomains"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=USER_ID,
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    'domain_name': 'payments.example.com',
    'pagination_cursor_ending_before': 'cursor_abc123',
    'fields_to_expand': ['webhooks', 'settings'],
    'limit_number_of_returned_objects': 25,
    'pagination_cursor_starting_after': 'cursor_def456',
    'include_enabled_domains': True
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))
